enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -357,6 +355,32 @@ func UploadTree(ctx context.Context, env environment.Env, dirHelper *DirHelper,
 				if err != nil {
 					return nil, err
 				}
+
+				ospb := &repb.OutputSymlink{
+					Path:   trimPathPrefix(fqfn, rootDir),
+					Target: target,
+				}
+				// In case the target is a dangling symlink (the target does not exist, or inaccessible),
+				// there is no way to tell whether it's a directory or a file symlink.
+				// Dangling symlinks could be used in packaging with targets that only exist on the
+				// deployment environments.
+				//
+				// Set both OutputFileSymlinks and OutputDirectorySymlinks only when the symlinks
+				// could be resolved.
+				if targetInfo, err := os.Stat(fqfn); err == nil {
+					// OutputFileSymlinks and OutputDirectorySymlinks are deprecated in REAPI v2.2,
+					// but Bazel still uses them as of v6.2.0.
+					//
+					// Set both the deprecated and the new fields to keep backward compatibility
+					// with older REAPI clients.
+					if targetInfo.IsDir() {
+						actionResult.OutputDirectorySymlinks = append(actionResult.OutputDirectorySymlinks, ospb)
+					} else {
+						actionResult.OutputFileSymlinks = append(actionResult.OutputFileSymlinks, ospb)
+					}
+				}
+				actionResult.OutputSymlinks = append(actionResult.OutputSymlinks, ospb)
```

Comment:
The documentation of `output_symlinks` says:

```
  // New in v2.1: this field will only be populated if the command
  // `output_paths` field was used, and not the pre v2.1 `output_files` or
  // `output_directories` fields.
```

So (1) based on this part: `this field will only be populated if the command `output_paths` field was used`

I think there should be a check around this line like

```
if IsOutputPath(fqfn) {
  actionResult.OutputSymlinks = append(actionResult.OutputSymlinks, ospb)
}
```

However, it looks like `IsOutputPath` is not technically checking the `action.OutputPaths` field. In workspace.go > SetTask, we are setting `outputPaths` to _either_ `output_paths` if set, else `concat(output_files, output_directories)`. This is pretty confusing IMO - we might want to refactor that. But for now we could maybe just pass in the `command` proto so that we have access to the original `output_paths` field.